src/App.tsx

```diff
@@ -1,19 +1,169 @@
 import React from 'react';
-import './App.css';
+import './App.scss';
 
 import users from './api/users';
+import todosFromApi from './api/todos';
+import { TodoList } from './components/TodoList';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+const preparedTodos: PrepearedTodo[] = todosFromApi.map(todo => ({
+  ...todo,
+  user: users.find(user => todo.userId === user.id) || null,
+}));
+
+type State = {
+  todos: PrepearedTodo[];
+  title: string;
+  userId: number;
+  hasTitleInputError: boolean;
+  hasUserSelectError: boolean;
+  hasWrongCharError: boolean;
 };
 
+class App extends React.PureComponent<{}, State> {
+  state: State = {
+    todos: preparedTodos,
+    title: '',
+    userId: 0,
+    hasTitleInputError: false,
+    hasUserSelectError: false,
+    hasWrongCharError: false,
+  };
+
+  inputChangeNameHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      title: event?.target.value,
+      hasTitleInputError: false,
+      hasWrongCharError: false,
+    });
+  };
+
+  selectUserHandler = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({
+      userId: Number(event?.target.value),
+      hasUserSelectError: false,
+    });
+  };
+
+  clearForm = () => {
+    this.setState({
+      title: '',
+      userId: 0,
+    });
+  };
+
+  validateForm = () => {
+    const { title, userId } = this.state;
+    const titleErr = /[a-zа-яA-ZА-ЯёЁ0-9]+$/.test(title);
+
+    if (!title || !userId || !titleErr) {
+      this.setState({
+        hasTitleInputError: !title,
+        hasUserSelectError: !userId,
+        hasWrongCharError: !titleErr,
+      });
+
+      return false;
+    }
+
+    return true;
+  };
+
+  formSubmitHandler = (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const { userId, title, todos } = this.state;
+    const isFormValid = this.validateForm();
+
+    if (isFormValid) {
+      const ids: number[] = todos.map(todo => todo.id);
+
+      this.setState((prevState) => ({
+        todos: [...prevState.todos,
+          {
+            userId,
+            id: Math.max(...ids) + 1,
+            title,
+            completed: false,
+            user: users.find(user => userId === user.id) || null,
+          },
+        ],
+      }));
+      this.clearForm();
```

Comment:
```suggestion
      }));
      
      this.clearForm();
```

Reply:
I wouldn't clear form if some inputs isn't valid. As for me better to give possibility for editing of entered data)))

Reply:
I mean, just add break line)